### PR TITLE
portfolio project challenge page

### PR DESCRIPTION
**Pull request by laurosilvacom**
Worked with @zacjones93 to add a portfolio project challenge page for the Introduction to Cloudflare Workers course. This serves as a new pattern for further courses that have project challenges. 

It also adds a  portfolio project card to the course landing page. 

CleanShot 2021-02-02 at 01 21 21@2x

CleanShot 2021-02-02 at 00 59 57@2x

Happy Hot Dog - 1LweXxLwVT0J2

### CODE REVIEW

File: src/components/layouts/collection-page-layout.tsx [hunk 1]
```diff
@@ -469,6 +475,42 @@ const CollectionPageLayout: React.FunctionComponent<CoursePageLayoutProps> = ({
                   </div>
                 </div>
               )}
+              {quickFacts && (
```
Comment (zacjones93) [inline src/components/layouts/collection-page-layout.tsx:478]: This should be updated when we start sourcing from sanity

File: src/components/layouts/collection-page-layout.tsx [hunk 2]
```diff
@@ -432,8 +435,11 @@ const CollectionPageLayout: React.FunctionComponent<CoursePageLayoutProps> = ({
 
               <div className="pt-5 md:hidden block">
                 <Fresh freshness={freshness} />
+
+                <CourseProjectCard courseProject={courseProject} />
```
Comment (zacjones93) [inline src/components/layouts/collection-page-layout.tsx:439]: @laurosilvacom got the CourseProject call out working on the course landing page!

File: src/data/courseDependenciesData.js [hunk 3]
```diff
@@ -54,7 +61,7 @@ const courseDependencyData = (courseSlug) =>
             title: 'Develop a Serverless Backend using Node.js on AWS Lambda',
             byline: 'Nik Graf・15m・Course',
             image:
-              'https://app.egghead.io/lessons/aws-course-overview-develop-a-serverless-backend-using-node-js-on-aws-lambda',
+              'https://d2eip9sf3oo6c2.cloudfront.net/series/square_covers/000/000/212/full/Egh_NodeAWSServerless_Final.png',
```
Comment (zacjones93) [inline src/data/courseDependenciesData.js:64]: Good catch

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+import Markdown from 'react-markdown'
+import Topic from 'components/search/components/topic'
+import Image from 'next/image'
+import Link from 'next/link'
+
+type LandingProps = {
+  course: any
+}
+
+const landingPage: FunctionComponent<LandingProps> = () => {
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:12]: page data for the project page

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+          className="pb-12"
+          topic={{
+            name: 'cloudflare',
+            label: 'Project Brief',
+          }}
+        >
+          <Markdown className="prose dark:prose-dark pt-2 sm:text-base text-sm leading-normal text-gray-800 dark:text-gray-200 mt-0">
+            {`You are a developer for a national concert promoter.
+
+Music fans from all over the world visit your site to see when and where the next show is going to be. To save them time when they visit the page, we want to show them upcoming events at a venue close to their location.
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:100]: you have to have your markdown start at the beginning of the file in your editor for this to behave as expected. This confused me for a long time...

Looks weird but once you get used to it, it's pretty nice

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+              stroke="#B0B0B0"
+              strokeWidth="7"
+              strokeDasharray="10 10"
+            />
+          </svg>
+        </div>
+
+        <div className="grid grid-cols-1 md:grid-cols-3 mt-16 mb-16">
+          <div className="relative px-10 py-10 bg-white dark:border-gray-800 dark:bg-gray-800 dark:text-gray-200 col-span-2 shadow rounded-md border border-gray-100 sm:mr-0 md:mr-4">
+            <div
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:128]: Took inspiration from the card borders from the other project page

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+              className="absolute rounded-t-lg rounded-b-none left-0 top-4 -mt-5 h-3 w-full bg-gradient-to-r"
+              style={{
+                background:
+                  'linear-gradient(to right, #F5C361 0%, #E75E3C 100%)',
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:132]: figma makes generating these gradients easy though you have to tweak the styles when using them in a React component

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+              Let us know what you build!
+            </p>
+
+            <a
+              className="inline-flex justify-center items-center px-6 py-4 font-semibold rounded-md bg-blue-600 text-white transition-all hover:bg-blue-700 ease-in-out duration-200 mt-12"
+              title="Share on twitter"
+              href="https://twitter.com/intent/tweet?text=I%20created%20my%20portfolio%20project%20for%20the%20Introduction%20to%20Cloudflare%20Workers%20course%20on%20@eggheadio!"
+              rel="noopener"
+            >
+              Tweet @eggheadio
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:252]: tweeting at egghead seems like a good CTA for now?

File: src/pages/projects/introduction-to-cloudflare-workers/index.tsx [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+  topic: Topic
+  className: any
+  pageData?: any
+  CTAComponent?: React.FC
+  ogImage?: string
+  verticalImage?: string
+  cardType?: CARD_TYPES
+}
+
+const ProjectBrief: React.FC<ProjectBriefProps> = ({
```
Comment (zacjones93) [inline src/pages/projects/introduction-to-cloudflare-workers/index.tsx:282]: This is the component from the default topic landing pages. Love the generated og image

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**